### PR TITLE
add support for OSLog API

### PR DESCRIPTION
**Pull request by WillSuo-Github**
HI everyone:
Add support for OSLog API.
Add critical and fault log level to match the new OSLog API.

### CODE REVIEW

**Comment by skreutzberger** [comment]
very cool, thanks for the PR, sorry for the delay, somehow I was not informed about it. Do you still want to have it applied?

**Comment by WillSuo-Github** [comment]
Thank you very much for your response. Yes, I would definitely like to merge the PR and make this repository support the OSLog API. However, I noticed that your repository can be used and compiled on multiple platforms, and I'm not entirely sure if the quality of my PR meets the requirements. Please review it carefully, and if it's possible to merge, it would be greatly appreciated.

**Comment by skreutzberger** [comment]
Thanks! I will hav a look at the end of the week.

**Comment by richardtop** [comment]
Hi, I'm interested in this PR merged as well, as in fact, I'm using a similar implementation. This PR will effectively push this implementation into SwiftyBeaver.

**Comment by skreutzberger** [comment]
thanks! I just released 2.0.1 with the privacy manifest update. Once that is well received I will release 2.1.0 with your OSLog API support. Expect it within the next 5 days :)

**Comment by manuelsalinas-mx** [comment]
@skreutzberger @richardtop @WillSuo-Github 
Apart of this, **Privacy Manifest** is required for Apple Store now.  any idea ?

**Comment by skreutzberger** [comment]
@manuelsalinas-mx the latest SwiftyBeaver version has the privacy manifest included.

File: Sources/BaseDestination.swift [hunk 1]
```diff
@@ -285,9 +292,11 @@ open class BaseDestination: Hashable, Equatable {
...
             str = levelString.error
 
-        default:
-            // Verbose is default
-            str = levelString.verbose
+        case .critical:
+            str = levelString.critical
+
+        case .fault:
+            str = levelString.fault
```
Comment (skreutzberger) [inline Sources/BaseDestination.swift:299]: can you change it so that we have verbose as default?

File: Sources/BaseDestination.swift [hunk 2]
```diff
@@ -309,8 +321,11 @@ open class BaseDestination: Hashable, Equatable {
         case .error:
             color = levelColor.error
 
-        default:
-            color = levelColor.verbose
+        case .critical:
+            color = levelColor.critical
+            
+        case .fault:
```
Comment (skreutzberger) [inline Sources/BaseDestination.swift:327]: what means fault? is it the highest severity?

**Comment by skreutzberger** [comment]
thanks! it will be released in some minutes

File: Sources/BaseDestination.swift (hunk 2 above)
Comment (WillSuo-Github) [reply Sources/BaseDestination.swift:327]: Critical should be the highest level, fault is just below critical. The official documentation from Apple is here https://developer.apple.com/documentation/os/logger.

**Comment by skreutzberger** [comment]
feel free to open a PR to better explain how your added functionality works to the README.

**Comment by manuelsalinas-mx** [comment]
Thanks for clarification @skreutzberger !  🥇

**Pull request merged**